float32 ball_x
float32 ball_y
bool    ball_on_ground
float32[] blocks_x
float32[] blocks_y
float32[] blocks_theta
float32[] board_origin
float32[] board_x_unit
float32[] board_y_unit
int32[]   on_blocks        #blocks you are allowed to pick up
float32[] goal_center   # in terms of xy unit vectors of the board
float32   goal_theta
bool      defense
